blueprints/structural_sections/concrete/reinforced_concrete_sections/rectangular_rcs.py

```diff
@@ -0,0 +1,877 @@
+
+        Returns
+        -------
+        Stirrup
+            Newly created Stirrup
+        """
+        # initiate the stirrup
+        stirrup = Stirrup(
+            coordinates=coordinates,
+            diameter=diameter,
+            distance=distance,
+            material=material,
+            shear_check=shear_check,
+            torsion_check=torsion_check,
+            mandrel_diameter_factor=mandrel_diameter_factor,
+            anchorage_length=anchorage_length,
+            based_on_cover=based_on_cover,
+            relative_start_position=relative_start_position,
+            relative_end_position=relative_end_position,
+        )
+
+        # add the stirrup to the list of stirrups
+        self.stirrups.append(stirrup)
+
+        return stirrup
+
+    def add_stirrup_along_edges(
+        self,
+        diameter: MM,
+        distance: MM,
+        material: ReinforcementSteelMaterial,
+        shear_check: bool = True,
+        torsion_check: bool = True,
+        mandrel_diameter_factor: float | None = None,
+        anchorage_length: MM = 0.0,
+        relative_start_position: DIMENSIONLESS = 0.0,
+        relative_end_position: DIMENSIONLESS = 1.0,
+    ) -> Stirrup:
+        """Add stirrups to the reinforced cross-section based on the present covers.
+
+        (No coordinates are needed for the creation of this stirrup.)
+
+        Parameters
+        ----------
+        diameter: MM
+            Diameter of the stirrups [mm].
+        distance: MM
+            Longitudinal distance between stirrups [mm].
+        material : ReinforcementSteelMaterial
+            Representation of the properties of reinforcement steel suitable for use with NEN-EN 1992-1-1
+        shear_check: bool
+            Take stirrup into account in shear check
+        torsion_check: bool
+            Take stirrup into account in torsion check
+        mandrel_diameter_factor: DIMENSIONLESS
+            Inner diameter of mandrel as multiple of stirrup diameter [-]
+            (default: 4⌀ for ⌀<=16mm and 5⌀ for ⌀>16mm) Tabel 8.1Na NEN-EN 1992-1-1 Dutch National Annex.
+        anchorage_length: MM
+            Anchorage length [mm]
+        relative_start_position: DIMENSIONLESS
+            Relative position of the start of the rebar in the x-direction of the host cross-section [-]
+        relative_end_position: DIMENSIONLESS
+            Relative position of the end of the rebar in the x-direction of the host cross-section [-]
+
+        Returns
+        -------
+        Stirrup
+            Newly created Stirrup
+        """
+        _max_x = max(x for x, _ in self.cross_section.vertices)
+        _min_x = min(x for x, _ in self.cross_section.vertices)
+        _min_y = min(y for _, y in self.cross_section.vertices)
+        _max_y = max(y for _, y in self.cross_section.vertices)
+
+        _left_bottom_corner = Point(_min_x + self.covers.left + (diameter / 2), _min_y + self.covers.lower + (diameter / 2))
```

Comment:
this means that the stirrup ALWAYs is the outer layer of rebar, most of the times that is the case, but not always.